### DISCUSSION TITLE
Badges in tables and data grids

### DISCUSSION THREAD

**Discussion by char-mainewang** [twilio-labs/paste#3203] [Q&A] [answered]
Hey Paste team, I'm in the process of migrating from evergreen to paste and have a few questions about how to use the Table and Datagrid component

**1. Badges in table and datagrid column headers**
The Paste Table guidance said A `Th` can be composed to include [icon buttons](https://paste.twilio.design/components/button#link-style-button). for RETL, we currently use a badge to call out which column is being used as the unique identifier (it's being used to determine if a record is new, updated or deleted). Does the Paste table / Data grid table header support badges or should we stick to the info icon

<img width="1109" alt="image" src="https://user-images.githubusercontent.com/87334135/234698497-447eacee-a01a-4f55-bf3b-06c09af5a39c.png">

**2. Min/Max column width**
RETL customers might be passing object / array data which could result in a super long string, is there a min/max cell width we should follow? An example here would be the RETL model's table, our customers might be looking to send order completed events to their destination, where each row would be a completed order. Sometimes our customers would have a product array column, where each {} is a different product their customer has purchased in the same order and each product has properties like name, id, price, sku, etc associated with it

<img width="940" alt="image" src="https://user-images.githubusercontent.com/87334135/234708977-bbc44ae4-2b15-471e-a67e-e885bc44dabf.png">

**Comment by SiTaggart** [comment, marked as answer]
Hi @char-mainewang 

For question one, yup, you can essentially put what ever you like (within reason) to a column header. Try to keep them simple, and not overload them, but it'll take a button, a badge, a link, an icon, anything kind of small and inline.

For question two, no, no real guidance from us. We try to be flexible and not too opinionated to allow product teams make the best decision that is relevant for displaying their data. It would be very difficult for us to create a catch all guideline that pleases everyone, and we'd rather empower you to make good design decisions for your product and your data.

Hope that clarifies a few things.

**Reply by char-mainewang** [reply]
Super helpful, thank you!
